test/SlowTests/Issues/RavenDB-14724.cs

```diff
@@ -42,14 +42,14 @@ public async Task DeleteDocumentAndRevisions()
                     var revisions = await session.Advanced.Revisions.GetForAsync<User>(id);
                     Assert.Equal(2, revisions.Count);
 
+                    session.Delete(id);
+                    await session.SaveChangesAsync();
+
```

Comment:
The original test wasn't failing because the revisions deletion behavior was wrong

Reply:
in what sense?

Reply:
in the original test, you delete a doc with revisions and you expect the revisions will be deleted too.
This assumption is wrong because when you `put` or `delete` a doc that has revisions (not conflicted) and doesn't have relevant revisions-config (collection or default), the revisions won't be touched.